src/app/store/technical-records/effects/technical-record-service.effects.spec.ts

```diff
@@ -515,5 +515,81 @@ describe('TechnicalRecordServiceEffects', () => {
     });
   });
 
-  // TODO: Unit tests for putUpdate and postProvisional methods
+  describe('updateTechnicalRecord$', () => {
+    it('should return a technical record on successfull API call', () => {
+      testScheduler.run(({ hot, cold, expectObservable }) => {
+        const technicalRecord = mockVehicleTechnicalRecordList();
+
+        // mock action to trigger effect
+        actions$ = hot('-a--', { a: updateTechRecords });
+
+        // mock service call
+        jest.spyOn(technicalRecordService, 'putUpdateTechRecords').mockReturnValue(cold('--a|', { a: technicalRecord[0] }));
+
+        // expect effect to return success action
+        expectObservable(effects.updateTechnicalRecord$).toBe('---b', {
+          b: updateTechRecordsSuccess({ vehicleTechRecords: technicalRecord })
+        });
+      });
+    });
+
+    it('should return an error message if not not found', () => {
+      testScheduler.run(({ hot, cold, expectObservable }) => {
+        // mock action to trigger effect
+        actions$ = hot('-a--', { a: updateTechRecords });
+
+        // mock service call
+        const expectedError = new HttpErrorResponse({
+          status: 500,
+          statusText: 'Internal server error'
+        });
+        jest.spyOn(technicalRecordService, 'putUpdateTechRecords').mockReturnValue(cold('--#|', {}, expectedError));
+
+        expectObservable(effects.updateTechnicalRecord$).toBe('---b', {
+          b: updateTechRecordsFailure({
+            error: 'Unable to update technical record null'
+          })
+        });
+      });
+    });
+  });
+
+  describe('postProvisionalTechRecord$', () => {
+    it('should return a technical record on successfull API call', () => {
+      testScheduler.run(({ hot, cold, expectObservable }) => {
+        const technicalRecord = mockVehicleTechnicalRecordList();
+
+        // mock action to trigger effect
+        actions$ = hot('-a--', { a: createProvisionalTechRecord });
+
+        // mock service call
+        jest.spyOn(technicalRecordService, 'postProvisionalTechRecord').mockReturnValue(cold('--a|', { a: technicalRecord[0] }));
+
+        // expect effect to return success action
+        expectObservable(effects.postProvisionalTechRecord).toBe('---b', {
+          b: createProvisionalTechRecordSuccess({ vehicleTechRecords: technicalRecord })
+        });
+      });
+    });
+
+    it('should return an error message if not not found', () => {
```

Comment:
Is there a reason the error codes are different between the two calls? 500 for update and 400 for post provisional?

Reply:
Yes, I didn't push save when testing something, thank you!